src/go/cmd/chart-assignment-controller/main.go

```diff
@@ -111,12 +120,20 @@ func setupAppV2(cfg *rest.Config, cluster string) error {
 	if err := chartassignment.Add(mgr, cluster); err != nil {
 		return errors.Wrap(err, "add ChartAssignment controller")
 	}
+
 	if *webhookEnabled {
-		srv := mgr.GetWebhookServer()
-		srv.CertDir = *certDir
+		if *cloudCluster {
+			srv := mgr.GetWebhookServer()
+			srv.CertDir = *certDir
 
-		webhook := chartassignment.NewValidationWebhookForEdgeCluster(mgr, cluster)
-		srv.Register("/chartassignment/validate", webhook)
+			srv.Register("/chartassignment/validate", chartassignment.NewValidationWebhook(mgr))
+		} else {
+			srv := mgr.GetWebhookServer()
+			srv.CertDir = *certDir
+
+			webhook := chartassignment.NewValidationWebhookForEdgeCluster(mgr, cluster)
+			srv.Register("/chartassignment/validate", webhook)
+		}
```

Comment:
this block has some code duplication: Maybe:
```go
var webhook *admission.Webhook
if *cloudCluster {
  webhook = chartassignment.NewValidationWebhook(mgr)
} else {
  webhook = chartassignment.NewValidationWebhookForEdgeCluster(mgr, cluster)
}
srv := mgr.GetWebhookServer()
srv.CertDir = *certDir
srv.Register("/chartassignment/validate", webhook)
```

Reply:
done